src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -23,17 +22,12 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
-            if (Type == EntityType.Subscription)
-            {
-                var entity = RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var otherEntity = other.RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var targetPathEquals = string.Equals(entity.Target.Path, otherEntity.Target.Path);
-                if (!targetPathEquals)
-                {
-                    return false;
-                }
-            }
-            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace);
+            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace) && DerivedEqual(other);
```

Comment:
Based on my conversation with @bording today, I have no objections to switching to an override of `Equals` (and `GetHashCode`) instead.

It seems the concern is API compatibility, Binary compatibility, and behavioral compatibility:

**API compatibility** is met by implementing `Equals` since a NuGet update and compile will still work without any user code changes needed..
**Binary compatibility** is met because the virtual methods are late bound.
**Behavioral compatibility** is met because `Equals` is currently a value comparison and this does not change that. It only corrects the value comparison to be correct for `SubscriptionInfo`. Implementing `GetHashCode` does not break our behavioral compatibility as `GetHashCode` [does not have behavioral guarantees](https://msdn.microsoft.com/en-us/library/system.object.gethashcode(v=vs.110).aspx#Anchor_1).